### PR TITLE
Setup circleci

### PR DESCRIPTION
**Pull request by tung-vu-td**
PR to setup Circle CI:
1. Use Circle CI config 2.1
2. Test ruby 2.5, 2.6, 2.7, 3.0 and jruby latest
3. Remove ruby tests for EOL 2.1 to 2.4
4. Small fixes to pass tests in Circle CI environments

### CODE REVIEW

**Review by kietdo360 — APPROVED** [review]
LGTM

**Review by nmpennypacker — APPROVED** [review]
Reviewed for Data Exfiltrations

**Comment by k0kubun** [comment]
How about removing Travis and AppVeyor configs? Both seem not working anyway, and you could test Windows in CircleCI as needed too.

File: td-client.gemspec [hunk 1]
```diff
@@ -26,6 +26,9 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency 'simplecov', '>= 0.5.4'
   gem.add_development_dependency 'rake'
   gem.add_development_dependency 'yard'
+  if RUBY_VERSION >= '3.0'
```
Comment (k0kubun) [inline td-client.gemspec:29]: This file is interpreted by a particular Ruby version on `gem push`, which means the dependency definition will be changed what Ruby version you'll use for `gem push`. 

You should simply declare the dependency regardless of the Ruby version if you really want to make sure it's specified as a development dependency when you use the gem from Ruby 3.0. Development dependencies on released packages probably don't really matter though.

Comment (tung-vu-td) [reply td-client.gemspec:29]: Ah I see. I have wrong understanding of RUBY_VERSION in gemspec. Thanks for catching it!

**Review by k0kubun — APPROVED** [review]
I think `win_ruby` could be a separate PR and it'd be nice to make master CI green first. But either way this is a great job.

**Pull request merged**